treeherder/webapp/api/push.py

```diff
@@ -70,6 +73,11 @@ def list(self, request, project):
                 )
 
             pushes = pushes.filter(repository=repository)
+    
+        if search_query:
+            pushes = pushes.annotate(
+                search_vector=SearchVector('commits__revision', 'commits__author', 'commits__comments')
+            ).filter(search_vector__icontains=search_query)
```

Comment:
use SearchQuery with the search_query as the first parameter, and `config="english"` as the second parameter